### PR TITLE
Issue 4  feature add min and max as properties to the base class

### PR DESCRIPTION
**Pull request by AndrewSazonov**
Thank you for your contribution to easyInterface. Please put your description here.

Before submitting this PR, please make sure:
* You include a description of the pull request
* Your code builds clean without any errors or warnings
* You are using approved terminology
* You have added unit tests

### CODE REVIEW

File: easyInterface/Diffraction/Interface.py [hunk 1]
```diff
@@ -331,6 +331,8 @@ def updatePhases(self):
         Synchronise the phases in project dictionary by queering the calculator object.
         """
         phases = self.calculator.getPhases()
+        self._log.debug(phases)
+        self._log.debug(self.project_dict['phases'])
```
Comment (wardsimon) [inline easyInterface/Diffraction/Interface.py]: Log should not be used this way. I'd suggest removal

File: easyInterface/Diffraction/Interface.py [hunk 2]
```diff
@@ -339,6 +341,8 @@ def updatePhases(self):
             self.project_dict.endBulkUpdate()
         else:
             k, v = self.project_dict['phases'].dictComparison(phases)
+            self._log.debug(k)
```
Comment (wardsimon) [inline easyInterface/Diffraction/Interface.py]: Log should not be used this way. I'd suggest removal

File: easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py [hunk 3]
```diff
@@ -1,10 +1,12 @@
+import numpy as np
 from copy import deepcopy
 from typing import Union, Optional, Any, NoReturn
 
 from easyInterface.Utils.units import Unit
 from easyInterface.Utils.DictTools import PathDict, UndoableDict
+from easyInterface import logger
 from easyInterface.Utils.Logging import logging
-from easyInterface import VERBOSE
+#from easyInterface import VERBOSE
```
Comment (wardsimon) [inline easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py]: Can be replaced with:
`from easyInterface import logger, VERBOSE`
I do not see logging being used.

File: easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py [hunk 4]
```diff
@@ -92,39 +94,57 @@ def __init__(self, value: Optional[Any] = None, unit: Optional[Union[str, Unit]]
...
     def __init__(self, value: object = None, unit: object = '') -> object:
         super().__init__(header='Undefined', tooltip='', url='', mapping=None, store=Data(value, unit))
+        self._log = logger.getLogger(__class__.__module__)
+        self.updateMinMax()
 
     def __repr__(self) -> str:
         return '{} {}'.format(self.value, self.getItemByPath(['store', 'unit']))
 
+    def updateMinMax(self):
+        self._log.debug(f"value: {self.value}")
```
Comment (wardsimon) [inline easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py]: debug or VERBOSE level? And, a more informative text

**Review by wardsimon — CHANGES_REQUESTED** [review]
Overall good and minor changes. 

My only comment is that log messages should be used for information instead of 'lazy debugging'

File: easyInterface/Diffraction/Interface.py (hunk 1 above)
Comment (AndrewSazonov) [reply easyInterface/Diffraction/Interface.py]: Debug logging is removed

File: easyInterface/Diffraction/Interface.py (hunk 2 above)
Comment (AndrewSazonov) [reply easyInterface/Diffraction/Interface.py]: Debug logging is removed

File: easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py (hunk 3 above)
Comment (AndrewSazonov) [reply easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py]: Debug logging is removed

File: easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py (hunk 4 above)
Comment (AndrewSazonov) [reply easyInterface/Diffraction/DataClasses/Utils/BaseClasses.py]: Debug logging is removed

**Pull request merged**